plugins/okta/komand_okta/util/api.py

```diff
@@ -32,6 +32,30 @@
 )
 
 
+def rate_limiting(max_tries: int):
+    def _decorate(func):
+        def _wrapper(*args, **kwargs):
+            retry = True
+            counter, delay = 0, 0
+            while retry and counter < max_tries:
+                if counter:
+                    time.sleep(delay)
+                try:
+                    retry = False
+                    return func(*args, **kwargs)
+                except ApiException as error:
+                    counter += 1
+                    delay = 2 ** (counter * 0.6)
+                    if error.cause == PluginException.causes[PluginException.Preset.RATE_LIMIT]:
+                        logging.info("Rate limiting error occurred. Retrying in %d seconds.", delay)
```

Comment:
Is it possible to use the logger instance from the plugin action/task/trigger/etc here? That logger instance uses structured logging which will make this appear nicer in our logging platform.

Reply:
Done 😄